packages/common/container-definitions/api-report/container-definitions.api.md

```diff
@@ -442,7 +445,7 @@ export interface IRuntime extends IDisposable {
     notifyOpReplay?(message: ISequencedDocumentMessage): Promise<void>;
     process(message: ISequencedDocumentMessage, local: boolean): any;
     processSignal(message: any, local: boolean): any;
-    setAttachState(attachState: AttachState.Attaching | AttachState.Attached): void;
+    setAttachState(attachState: typeof AttachState.Attaching | typeof AttachState.Attached): void;
```

Comment:
Does `typeof AttachState.Attaching` not resolve to "string"? or does it resolve to the string value in the const object?

Reply:
Yeah, I think the secret here is the `as const;` after the declaration, which makes the types of the object's properties not be "just some string", but "the specific string in the definition, and nothing else".